### PR TITLE
Maya/download modal refactor

### PR DESCRIPTION
**Pull request by mmmmmmaya**
### Summary
- **What:** Create a component for download modal options. 
- **Why:** It was super hard to parse style tags vs relevant content before. Also there were a couple styling bugs.

### Testing
1. Select a sample that failed genome recovery
1. Check out the download modal
2. Top option should be disabled and have tooltip on hover

### Demos
No visual changes.

### Checklist
- [x] I merged latest `trunk`
- [x] I manually verified the change
- [x] I added labels to my PR
- [x] I either asked design for a review or hid my changes behind a feature flag
- [ ] I tested in multiple browsers
- [ ] I added relevant unit tests
- [ ] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

### CODE REVIEW

File: src/frontend/src/views/Data/components/SamplesView/components/SampleTableModalManager/components/DownloadModal/components/DownloadMenuSelection/index.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        <StyledCheckbox
+          id={id}
+          disabled={isDisabled}
+          onChange={onChange}
+          stage={isChecked ? "checked" : "unchecked"}
+        />
+      </CheckBoxInfo>
+      <CheckboxLabel htmlFor={id}>
+        <span onMouseOver={(e) => setAnchorEl(e.currentTarget)}>
+          <DownloadType>{downloadTitle} </DownloadType> ({fileTypes})
```
Comment (vincent-czi) [inline src/frontend/src/views/Data/components/SamplesView/components/SampleTableModalManager/components/DownloadModal/components/DownloadMenuSelection/index.tsx:47]: Nit
```suggestion
          <DownloadType>{downloadTitle}</DownloadType> ({fileTypes})
```

**Review by vincent-czi — APPROVED** [review]

File: src/frontend/src/views/Data/components/SamplesView/components/SampleTableModalManager/components/DownloadModal/components/DownloadMenuSelection/index.tsx (hunk 1 above)
Comment (mmmmmmaya) [reply src/frontend/src/views/Data/components/SamplesView/components/SampleTableModalManager/components/DownloadModal/components/DownloadMenuSelection/index.tsx:47]: the space is necessary for separating the bold text from the regular weight text
